scripts/extract.py

```diff
@@ -44,19 +45,33 @@ def process(self):
         else:
             self.extract_single_process()
 
-        self.alignments.write_alignments(self.faces.faces_detected)
-
+        self.write_alignments()
         images, faces = Utils.finalize(self.images.images_found,
                                        self.faces.num_faces_detected,
                                        self.faces.verify_output)
         self.images.images_found = images
         self.faces.num_faces_detected = faces
 
+    def write_alignments(self):
+        self.alignments.write_alignments(self.faces.faces_detected)
+
+
     def extract_single_process(self):
         """ Run extraction in a single process """
+
+        alignments = self.alignments.read_alignments()
+        
+        frame_no = 0
         for filename in tqdm(self.images.input_images, file=sys.stdout):
+            basename = os.path.basename(filename)
+            if basename in alignments:
+                continue
             filename, faces = self.process_single_image(filename)
-            self.faces.faces_detected[os.path.basename(filename)] = faces
+            self.faces.faces_detected[basename] = faces
+            frame_no = frame_no + 1
+            if frame_no == self.save_interval:
```

Comment:
`if hasattr(self.args, "save_interval") and self.args.save_interval == frame_no`